test-cypress/steps/graphql/edit-graphql-endpoint-steps.js

```diff
@@ -0,0 +1,100 @@
+export class EditGraphqlEndpointSteps {
+    static getModal() {
+        return cy.get('.graphql-endpoint-configuration-modal');
+    }
+
+    static getModalTitle() {
+        return this.getModal().find('.modal-title');
+    }
+
+    static getDynamicForm() {
+        return this.getModal().find('dynamic-form');
+    }
+
+    static getFormFields() {
+        return this.getDynamicForm().find('.form-field');
+    }
+
+    static getInputField(index) {
+        return this.getFormFields().find('.input-field input').eq(index);
+    }
+
+    static getBooleanField(index) {
+        return this.getFormFields().find('.boolean-field input').eq(index);
+    }
+
+    static getSelectField(index) {
+        return this.getFormFields().find('.select-field select').eq(index);
+    }
+
+    static getMultiSelectField(index) {
+        return this.getFormFields().find('.multiselect-field multiselect-dropdown').eq(index);
+    }
+
+    static toggleMultiSelectOption(index, optionLabel) {
+        this.getMultiSelectField(index)
+            .within(() => {
+                cy.get('button.dropdown-toggle').click();
+                cy.get('ul.dropdown-menu li')
+                    .contains(optionLabel)
+                    .click();
+            });
+    }
+
+    static verifyMultiSelectOptionSelected(index, optionLabel) {
+        this.getModalTitle().click();
```

Comment:
Leaving the dropdown open after this might lead to flakiness if it happens to cover some filed below